### PR TITLE
Add user operation event listener for remove user role assignments on user deletion events

### PR DESCRIPTION
**Pull request by sadilchamishka**

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/listener/OrganizationUserOperationEventListener.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.wso2.carbon.identity.organization.management.role.management.service.listener;
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/listener/OrganizationUserOperationEventListener.java:1]: License?

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/listener/OrganizationUserOperationEventListener.java:19]: Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/198/commits/01e07826f5331bbdbad4d880520f8a16afcd149e

File: components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAO.java [hunk 2]
```diff
@@ -151,4 +151,6 @@ boolean checkRoleExists(String organizationId, String roleId, String roleName)
      */
     int getTotalOrganizationRoles(String organizationId, List<ExpressionNode> expressionNodes, List<String> operators)
             throws OrganizationManagementServerException;
+
+    void deleteUserRoleByUserId(String userId) throws OrganizationManagementServerException;
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAO.java:155]: add method comment.
Few suggestions to method name: `deleteUserFromOrganizationRoles`, `removeUserFromOrganizationRoles`, `unassignUserFromOrganizationRoles`

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAO.java]: - Added method comments.
- Accepted the `deleteUserFromOrganizationRoles` suggestion

Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/198/commits/b5d343f5c48850bd5ce8c861f727b4190526f4c8

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**